tests/partitions_test.py

```diff
@@ -275,6 +276,92 @@ def test_new_joiner_helps_liveness(network, args):
         network.wait_for_all_nodes_to_commit(primary=primary)
 
 
+@reqs.description("Test election while reconfiguration is in flight")
+@reqs.at_least_n_nodes(3)
+def test_election_reconfiguration(network, args):
+    # Test for issue described in https://github.com/microsoft/CCF/issues/3948
+    # Note: this test makes use of node-endorsed secondary RPC interface since
+    # new nodes never observe commit of their configuration and thus never
+    # open their service-endorsed primary RPC interface.
+    primary, backups = network.find_nodes()
+
+    LOG.info("Join new nodes without trusting them just yet")
+    new_nodes = []
+    # Start N+1 new nodes to make sure they cannot elect of them as a primary
+    # without approval from the original configuration
+    for _ in range(len(network.nodes) + 1):
+        rpc_interfaces = {
+            infra.interfaces.PRIMARY_RPC_INTERFACE: infra.interfaces.RPCInterface(
+                host="localhost"
+            )
+        }
+        rpc_interfaces.update(infra.interfaces.make_secondary_interface())
+        new_node = network.create_node(infra.interfaces.HostSpec(rpc_interfaces))
+        network.join_node(new_node, args.package, args, from_snapshot=False)
+        new_nodes.append(new_node)
+
+    LOG.info("Isolate original backups and issue reconfiguration of another quorum")
+    with network.partitioner.partition(backups):
+        LOG.info("Trust all new nodes in one single proposal")
+        # Note: Commit is stuck since a majority of backups in initial configuration
+        # are isolated
+        network.consortium.trust_nodes(
+            primary,
+            [n.node_id for n in new_nodes],
+            valid_from=datetime.utcnow(),
+            wait_for_global_commit=False,
+        )
+
+        for node in new_nodes:
+            node.wait_for_node_to_join(
+                interface_name=infra.interfaces.SECONDARY_RPC_INTERFACE
+            )
+            # Wait for configuration tx to be replicated to new node
+            network.wait_for_node_in_store(
+                node,
+                node.node_id,
+                ccf.ledger.NodeStatus.TRUSTED,
+                interface_name=infra.interfaces.SECONDARY_RPC_INTERFACE,
+            )
+
+        LOG.info(f"Stop primary node {primary.local_node_id} to trigger election")
+        primary.stop()
+
+        LOG.info(
+            "Make sure that new nodes cannot elect a primary node among themselves"
+        )
+        try:
+            network.wait_for_new_primary(
+                primary,
+                nodes=new_nodes,
+                interface_name=infra.interfaces.SECONDARY_RPC_INTERFACE,
+                timeout_multiplier=1,
```

Comment:
```suggestion
                timeout_multiplier=3,
```

Just a guess. A lot of our elections require multiple rounds of election time outs before they pass in good, expected situations. If we want to assert that no primary _can_ be elected, I think we need to wait for multiple elections.

Reply:
Good point. I wanted the test to run quickly as we're expecting no new primary to be elected but we should be consistent and wait for a reasonable amount of time here too.